### PR TITLE
Author Page

### PR DESCRIPTION
**Pull request by MartinAcob**

### CODE REVIEW

File: client/src/pages/lessons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+  const [sortDesc, setSortDesc] = React.useState(true);
+  const rowsPerPage = 10;
+
+  React.useEffect(() => {
+    let mounted = true;
+    fetchLessons(rowsPerPage, prevPages[page], sortBy, sortDesc)
+      .then((lessons) => {
+        console.log(`fetchLessons got`, lessons);
+        if (mounted) {
+          if (lessons !== undefined) {
```
Comment (beatthat) [inline client/src/pages/lessons/index.tsx:116]: prefer to just use js `falsy`ness for tests and also don't see a need for two separate `if` blocks here so all in all could be

```typescript
if (mounted && lessons) {
```

File: client/src/pages/lessons/index.tsx [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+            const tmp: any = lessons.edges;
+            tmp.map((lesson: any) => {
```
Comment (beatthat) [inline client/src/pages/lessons/index.tsx:118]: do we need this `tmp` var? why not just `lessons.edges.map(lesson =>` ?

File: client/src/pages/lessons/index.tsx [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+              lesson.node.updatedAt = new Date(lesson.node.updatedAt);
+            });
+            setLessons(lessons);
+          }
+        }
+      })
+      .catch((err) => console.error(err));
+    return () => {
```
Comment (beatthat) [inline client/src/pages/lessons/index.tsx:126]: what does this do? Does `useEffect` expect as return a function that in turn returns an object?

If too complicated to explain in writing, maybe just walk me through it over hangout call

File: client/src/pages/lessons/index.tsx [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+      mounted = false;
+    };
+  }, []);
+
+  React.useEffect(() => {
+    fetchLessons(rowsPerPage, prevPages[prevPages.length - 1], sortBy, sortDesc)
+      .then((lessons) => {
+        console.log(`page switch fetchLessons got`, lessons);
+        if (lessons !== undefined) {
```
Comment (beatthat) [inline client/src/pages/lessons/index.tsx:135]: simplify to `if(lessons)` this varies across languages, but in javascript even an empty array is `truthy`

File: client/src/pages/lessons/index.tsx [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+          const tmp: any = lessons.edges;
```
Comment (beatthat) [inline client/src/pages/lessons/index.tsx:136]: same q: why `tmp`?

File: client/src/pages/lessons/edit.tsx [hunk 6]
```diff
@@ -0,0 +1,579 @@
...
+  const [create, setCreate] = React.useState(false);
+  const [expectationOpen, setExpectationOpen] = React.useState(true);
+
+  React.useEffect(() => {
+    if (lessonId !== "new") {
+      let mounted = true;
+      fetchLesson(lessonId)
+        .then((lesson: Lesson) => {
+          console.log("fetchLesson got", lesson);
+          if (mounted) {
```
Comment (beatthat) [inline client/src/pages/lessons/edit.tsx:109]: simplify to `if(mounted && lesson)`

File: client/src/pages/lessons/edit.tsx [hunk 7]
```diff
@@ -0,0 +1,579 @@
...
+    setChange(true);
+    setLesson({ ...lesson, question: question });
+  }
+
+  function handleConclusionChange(con: string, index: number): void {
+    setChange(true);
+    const copyLesson = { ...lesson };
+    const copyConclusion = [...copyLesson.conclusion] as Array<any>;
+    copyConclusion[index] = con;
+    setLesson({ ...lesson, conclusion: copyConclusion });
```
Comment (beatthat) [inline client/src/pages/lessons/edit.tsx:151]: Possilby all these immutable copy/edits can be simplified with `Array.slice`, e.g.

```typescript
setLesson({ 
  ...lesson, 
  conclusion: [
    ...lesson.conclusion.slice(0, index), 
    con, 
    ...lesson.conclusion.slice(index + 1)
  ]
});
```

?

File: client/src/pages/lessons/index.tsx (hunk 3 above)
Comment (MartinAcob) [reply client/src/pages/lessons/index.tsx:126]: Without the mounted, I get a warning for react on console about unmounted components and potential memory leak. This is one of the solutions I found online.

**Pull request merged**